web/pages/api/v0/bet/index.ts

```diff
@@ -4,11 +4,16 @@ import {
   CORS_ORIGIN_LOCALHOST,
 } from 'common/envs/constants'
 import { applyCorsHeaders } from 'web/lib/api/cors'
+import { assertHTTPMethod } from 'web/lib/api/validation'
 import { fetchBackend, forwardResponse } from 'web/lib/api/proxy'
 
 export const config = { api: { bodyParser: false } }
 
 export default async function route(req: NextApiRequest, res: NextApiResponse) {
+  if (!assertHTTPMethod(req, res, 'POST')) {
+    return
+  }
```

Comment:
I don't believe we should need these checks on the routes that just proxy to Firebase, because the Firebase functions should be checking the method and forwarding back appropriate error messages and status codes. Have you found otherwise?

Reply:
Currently the error message for proxied requests is suboptimal:

```
curl -i -X GET https://manifold.markets/api/v0/market
HTTP/2 500
content-type: application/json; charset=utf-8
x-matched-path: /api/v0/market
cache-control: public, max-age=0, must-revalidate
content-length: 47
date: Thu, 16 Jun 2022 23:39:49 GMT
etag: "2f-NLzThRQiuTD7mRueNLeWmzSmcSA"
vary: Origin
x-vercel-cache: MISS
age: 0
server: Vercel
x-vercel-id: pdx1::cle1::4qt5v-1655422789083-3869a3577425
strict-transport-security: max-age=63072000

{"message":"Error communicating with backend."}
```

With the change, it looks like:
```
HTTP/1.1 405 Method Not Allowed
Allow: POST
Date: Thu, 16 Jun 2022 23:40:14 GMT
Connection: keep-alive
Keep-Alive: timeout=5
Transfer-Encoding: chunked

Method GET Not Allowed
```

Reply:
That's on me then, I will look and understand what is going on. The goal was for the Vercel routes to be a very transparent proxy -- having logic on both sides is a recipe for confusion.

Reply:
Fixed in https://github.com/manifoldmarkets/manifold/pull/542.

Reply:
Thanks!